### PR TITLE
Fix for #102

### PR DESCRIPTION
**Pull request by sindhus**

### CODE REVIEW

File: baseframe/static/js/baseframe.js [hunk 1]
```diff
@@ -1,10 +1,6 @@
 // This is a global function. Isn't there a better way to do this?
 
 function activate_widgets() {
-  // Activate select2.js and CodeMirror for non-mobile browsers
-  if (!navigator.userAgent.match(/(iPod|iPad|iPhone|Android)/)) {
-    $('select:not(.notselect)').select2({allowClear: true});
-
```
Comment (jace) [inline baseframe/static/js/baseframe.js]: Now you've activated Select2 on mobile as well, which is not wanted. Only move CodeMirror out.

File: baseframe/Makefile [hunk 2]
```diff
@@ -3,7 +3,8 @@ all: tinymce codemirror
 
 tinymce3_zip = tinymce_3.5.11_jquery.zip
 tinymce4_zip = tinymce_4.1.7_jquery.zip
-codemirror_zip = codemirror.zip
+codemirror4_zip = 4.11.0.zip
+codemirror5_zip = 5.8.0.zip
```
Comment (jace) [inline baseframe/Makefile]: We don't need both 4 and 5 (unlike TinyMCE 3 and 4, which represent an incompatible upgrade).

File: baseframe/Makefile [hunk 3]
```diff
@@ -19,13 +20,26 @@ tinymce:
 	rm -rf $(tinymce4_zip) tinymce
 
 codemirror:
-	curl -L -O http://codemirror.net/$(codemirror_zip)
+	curl -L -O https://github.com/codemirror/CodeMirror/archive/$(codemirror4_zip)
```
Comment (jace) [inline baseframe/Makefile]: Don't we want 5?

File: baseframe/Makefile [hunk 4]
```diff
@@ -19,13 +20,26 @@ tinymce:
...
 	unzip $(codemirror_zip)
+	# Remove old unversioned codemirror
```
Comment (jace) [inline baseframe/Makefile]: This comment is not needed.

File: baseframe/Makefile [hunk 5]
```diff
@@ -19,13 +20,26 @@ tinymce:
...
-	mv codemirror-*/mode static/js/codemirror
-	mv codemirror-*/theme static/js/codemirror
-	rm -rf $(codemirror_zip) codemirror-*
+	mkdir static/js/codemirror4
+	mv codemirror-*/addon static/js/codemirror4
+	mv codemirror-*/keymap static/js/codemirror4
+	mv codemirror-*/lib static/js/codemirror4
+	mv codemirror-*/mode static/js/codemirror4
+	mv codemirror-*/theme static/js/codemirror4
+	rm -rf $(codemirror4_zip) codemirror4-*
```
Comment (jace) [inline baseframe/Makefile]: Don't install CodeMirror 4 if 5 is backward compatible.

File: baseframe/Makefile [hunk 6]
```diff
@@ -19,13 +20,26 @@ tinymce:
...
+
+	curl -L -O https://github.com/codemirror/CodeMirror/archive/$(codemirror5_zip)
+	unzip $(codemirror_zip)
+	mkdir static/js/codemirror5
```
Comment (jace) [inline baseframe/Makefile]: Put these in the `codemirror` folder, not `codemirror5` (unless backward incompatible).

File: baseframe/static/js/baseframe.js [hunk 7]
```diff
@@ -1,67 +1,10 @@
...
-    $('textarea.stylesheet').each(function() {
-      var editor = CodeMirror.fromTextArea(this, cm_css_config);
-      var delay;
-      editor.on('change', function(instance){
-        clearTimeout(delay);
-        delay = setTimeout(function() {
-          editor.save();
-        }, 300);
-      });
-    });
```
Comment (jace) [inline baseframe/static/js/baseframe.js]: This code should be moved out of the `if` block, not removed.

File: baseframe/static/js/baseframe.js [hunk 8]
```diff
@@ -1,17 +1,18 @@
...
-      viewportMargin: Infinity,
-      extraKeys: {
+  var cm_markdown_config = { mode: 'gfm',
+    lineNumbers: false,
+    theme: "default",
+    lineWrapping: true,
+    autoCloseBrackets: true,
+    viewportMargin: Infinity,
+    extraKeys: {
         "Enter": "newlineAndIndentContinueMarkdownList",
```
Comment (jace) [inline baseframe/static/js/baseframe.js]: Too much indentation here.

**Comment by jace** [comment]
Have we tested CodeMirror 5.x on mobile? Funnel's proposal form used to be broken on Chrome Android under 4.x.

**Comment by sindhus** [comment]
@kiran As asked, I checked copy and paste on my Nexus 5 running Android 6.0 and it works fine. I was also able to to create and edit a proposal (created org, team etc). However I don't know how to test local app running on my computer on my phone.

**Comment by jace** [comment]
@sindhus What did you test if it wasn't the local app running CodeMirror 5.x? `talkfunnel.com` currently has CodeMirror disabled on mobile.

**Comment by sindhus** [comment]
@kiran you said "Point your phone to CodeMirror’s site and confirm you can use it like a regular text field.". I figured CodeMirror website itself will be running latest version of CM and so I opened up codemirror.net and tried the editor there.

**Comment by jace** [comment]
I meant "I was also able to to create and edit a proposal". Did you mean that in your browser and not your phone?

File: baseframe/Makefile [hunk 9]
```diff
@@ -19,13 +19,13 @@ tinymce:
...
-	mv codemirror-*/lib static/js/codemirror
-	mv codemirror-*/mode static/js/codemirror
-	mv codemirror-*/theme static/js/codemirror
-	rm -rf $(codemirror_zip) codemirror-*
+	mv CodeMirror-$(codemirror)/addon static/js/codemirror
+	mv CodeMirror-$(codemirror)/keymap static/js/codemirror
+	mv CodeMirror-$(codemirror)/lib static/js/codemirror
+	mv CodeMirror-$(codemirror)/mode static/js/codemirror
+	mv CodeMirror-$(codemirror)/theme static/js/codemirror
+	rm -rf $(codemirror) CodeMirror-$(codemirror)
```
Comment (jace) [inline baseframe/Makefile]: This should be `rm -rf $(codemirror).zip CodeMirror-$(codemirror)`

File: baseframe/Makefile [hunk 10]
```diff
@@ -3,7 +3,7 @@ all: tinymce codemirror
 
 tinymce3_zip = tinymce_3.5.11_jquery.zip
 tinymce4_zip = tinymce_4.1.7_jquery.zip
-codemirror_zip = codemirror.zip
+codemirror = 5.8.0
```
Comment (jace) [inline baseframe/Makefile]: To avoid confusion, this variable should be named `codemirror_ver`

**Comment by sindhus** [comment]
@kiran On my computer of course.

I said I tested the ability to add/edit proposal on the local funnel running on my computer.
I don't know how to access local funnel running on my computer on my phone.

File: baseframe/Makefile (hunk 10 above)
Comment (sindhus) [inline baseframe/Makefile]: fixed.

File: baseframe/Makefile (hunk 9 above)
Comment (sindhus) [inline baseframe/Makefile]: fixed.

**Comment by jace** [comment]
Previous sentence in that comment said "Android". Next sentence referring to "desktop" wasn't obvious. :)

**Comment by kiran** [comment]
I don't think you meant to tag @kiran.

**Comment by jace** [comment]
@kiran Apologies for pulling you into this. There doesn't seem to be a way to untag you, but FWIW this is closed now.

**Comment by sindhus** [comment]
@jace have removed the tag. since i wrote the comment, only I can untag.

**Pull request merged**